### PR TITLE
creating boilerplate for OpenSearch Source

### PR DESCRIPTION
**Pull request by cmanning09**
### Description
Creating a high level class structure. The goal is to align on plugin structure and make it easier for multiple contributors to work on the plugin at once. This PR does not include unit tests as there are no methods implemented at this time.

#### Proposal
This is the current proposal for the class design. This is a rough sketch I threw together. It captures the classes in this PR and helps demonstrate the relationships between them:

 
### Issues Resolved
related to #1985 
 
### Check List
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package org.opensearch.dataprepper.plugins.source.opensearch.worker.client;
+
+import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSourceConfiguration;
+
+/**
+ * SearchAccessorStrategy determines which {@link SearchAccessor} (Elasticsearch or OpenSearch) should be used based on
+ * the {@link OpenSearchSourceConfiguration}.
+ * @since 2.4
+ */
+public class SearchAccessorStrategy {
```
Comment (graytaylor0) [inline data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java:14]: Is this class replacing the `SearchClientFactory` from the class diagram?

Comment (cmanning09) [reply data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java:14]: Yes, I forgot I renamed it. Good catch.

File: data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/model/CreatePitRequest.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model;
+
+public class CreatePitRequest {
```
[the remark this answers is not in the document]
Comment (cmanning09) [reply data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/model/CreatePitRequest.java:7]: ElastichSerach and OpenSearch requests and responses may differ slightly in the future. This model is an attempt to decouple that. From these models we can convert them into either payload structure more easily in the future.

[the remark this answers is not in the document]
Comment (cmanning09) [reply data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/model/CreatePitRequest.java:7]: I am open to removing these and adding as needed if you'd prefer.

[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/model/CreatePitRequest.java:7]: Makes sense. I think leaving it in this PR is fine

**Review by graytaylor0 — APPROVED** [review]

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**